src/Domain/PasswordResetToken.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace BigGive\Identity\Domain;
+
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\JoinColumn;
+use Symfony\Component\Uid\Uuid;
+
+/**
+ * @ORM\Entity(repositoryClass="BigGive\Identity\Repository\PasswordResetTokenRepository")
+ * @ORM\Table(name="PasswordResetToken", indexes={
+ *     @ORM\Index(name="secret", columns={"secret"}),
+ * })
+ */
+class PasswordResetToken
+{
+    use TimestampsTrait;
+
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="uuid", unique=true)
+     * @ORM\GeneratedValue(strategy="CUSTOM")
+     * @ORM\CustomIdGenerator(class="\Symfony\Bridge\Doctrine\IdGenerator\UuidGenerator")
+     */
+    private ?Uuid $id = null;
+
+    /**
+     * @ORM\Column(type="uuid", unique=true)
+     * @ORM\CustomIdGenerator(class="\Symfony\Bridge\Doctrine\IdGenerator\UuidGenerator")
+     */
+    private readonly Uuid $secret;
+
+    /**
+     * @ORM\Column(type="uuid", unique=true)
+     * @ManyToOne(targetEntity="Person")
+     * @JoinColumn(name="address_id", referencedColumnName="id")
+     */
+    public readonly Person $person;
+
+    /**
+     * @ORM\Column(type="datetime", nullable=true)
+     */
+    private ?\DateTimeImmutable $used = null;
```

Comment:
for some reason Doctrine is trying to make this non-null in the DB schema. I manually edited the migration file to make it nullable but that's no good really, need to get the diff to do nothing.